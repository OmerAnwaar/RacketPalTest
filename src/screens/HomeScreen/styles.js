import {StyleSheet} from 'react-native';
import {colors, theme} from '../../shared/constants';
const styles = StyleSheet.create({
  container: {
    flex: 1,
    ...theme.CENTER,
  },
  button: {
    padding: 15,
    borderRadius: 10,
    backgroundColor: colors.BLUE,
    ...theme.CENTER,
  },
  textStyle: {
    color: colors.WHITE,
    fontWeight: 'bold',
  },
});

export default styles;
