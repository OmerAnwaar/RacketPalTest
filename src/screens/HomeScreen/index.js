import React, {useState} from 'react';
import ReviewModal from '../../modals/ReviewModal';
import {Text, TouchableOpacity, View} from 'react-native';
import styles from './styles';
import {logAnalytic} from 'shared/utils';

const HomeScreen = () => {
  const [modal, setModal] = useState(false);

  function toggleModal() {
    setModal(prevState => !prevState);
  }

  function onDisplay() {
    logAnalytic('review-modal', '123453');
    setModal(true);
  }

  return (
    <View style={styles.container}>
      <TouchableOpacity
        testID="rating"
        onPress={onDisplay}
        style={styles.button}>
        <Text style={styles.textStyle}>Give a Feedback</Text>
      </TouchableOpacity>
      <ReviewModal visible={modal} toggleModal={toggleModal} />
    </View>
  );
};

export default HomeScreen;
