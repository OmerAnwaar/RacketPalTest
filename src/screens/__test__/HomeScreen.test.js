import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from '../HomeScreen';

const mockedFirebaseAnalyticsLogEvent = jest.fn();
const mockedFirebaseAnalyticsLogSelectContent = jest.fn();

jest.mock('@react-native-firebase/analytics', () => () => {
  return {
    logEvent: mockedFirebaseAnalyticsLogEvent,
    logSelectContent: mockedFirebaseAnalyticsLogSelectContent,
  };
});
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

describe('Test Screen', () => {
  it('renders Home screen correctly', () => {
    const snapshot = renderer.create(<HomeScreen />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  // it('onToggle coverage', () => {
  //   // render the component on virtual dom
  //   let tree = renderer.create(<HomeScreen />).toJSON();
  //   expect(global.utils.findById(tree, 'rating')).toBeDefined();
  // });
});
