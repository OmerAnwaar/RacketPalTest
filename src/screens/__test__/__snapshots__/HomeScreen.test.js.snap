// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Screen renders Home screen correctly 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#2196f3",
        "borderRadius": 10,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 15,
      }
    }
    testID="rating"
  >
    <Text
      style={
        {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Give a Feedback
    </Text>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  />
</View>
`;
